pulp_ansible/tests/functional/api/collection/test_sync.py

```diff
@@ -137,3 +137,55 @@ def test_mirror_sync(self):
         # Step 5
         self.assertDictEqual(get_content_summary(repo), ANSIBLE_COLLECTION_FIXTURE_SUMMARY)
         self.assertDictEqual(get_removed_content_summary(repo), ANSIBLE_FIXTURE_CONTENT_SUMMARY)
+
+    def test_mirror_sync_with_requirements(self):
+        """Sync multiple plugin into the same repo with mirror as `True`,
+        using requirements.yml
+
+        This test targets the following issue: 5250
+
+        * `<https://pulp.plan.io/issues/5250>`_
+
+        This test does the following:
+
+        1. Create a repo.
+        2. Create two remotes
+            a. Role remote
+            b. Collection remote
+        3. Sync the repo with Role remote.
+        4. Sync the repo with Collection remote with ``Mirror=True``.
+        5. Verify whether the content in the latest version of the repo
+            has only Collection content and Role content is deleted.
+        """
+        # Step 1
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo["_href"])
+
+        # Step 2
+        role_remote = self.client.post(ANSIBLE_REMOTE_PATH, gen_ansible_remote())
+        self.addCleanup(self.client.delete, role_remote["_href"])
+
+        for requirement_file in create_requirement_file():
+            collection_remote = self.client.post(
+                ANSIBLE_COLLECTION_REMOTE_PATH, gen_ansible_remote(
+                    url=ANSIBLE_COLLECTION_FIXTURE_URL,
+                    requirements_file=requirement_file,
```

Comment:
Test is always breaking here:
- if string: `b'{"requirements_file":["The submitted data was not a file. Check the encoding type on the form."]}'`
- if file: `TypeError: Object of type TextIOWrapper is not JSON serializable`